Source/BugsnagEvent.m

```diff
@@ -109,17 +109,23 @@ id BSGLoadConfigValue(NSDictionary *report, NSString *valueName) {
     ?: [report valueForKeyPath:fallbackKeypath]; // some custom values are nested
 }
 
+/**
+ * Attempt to find a context (within which the event is being reported)
+ * This can be found in user-set metadata of varying specificity or the global
+ * configuration.  Returns nil if no context can be found.
+ *
+ * @param report A dictionary of report data
+ * @param metadata Additional relevant data
+ * @returns A string context if found, or nil
+ */
 NSString *BSGParseContext(NSDictionary *report, NSDictionary *metadata) {
-    id context = [report valueForKeyPath:@"user.overrides.context"];
-    if ([context isKindOfClass:[NSString class]])
-        return context;
-    context = metadata[BSGKeyContext];
-    if ([context isKindOfClass:[NSString class]])
-        return context;
-    context = BSGLoadConfigValue(report, @"context");
-    if ([context isKindOfClass:[NSString class]])
-        return context;
-    return nil;
+    // Select the first non-nil value, in order of specificity
+    id context = [report valueForKeyPath:@"user.overrides.context"]
+        ?: metadata[BSGKeyContext]
+        ?: BSGLoadConfigValue(report, @"context")
+        ?: [[Bugsnag configuration] context];
```

Comment:
This change is a bit tricky since there are two different ways an Event can be constructed.

1. `notify()` was just called and the Event instance has no persisted file. In this case, loading `context` from Configuration makes sense.
2. A persisted report was read from disk and constructed an Event object. In this case, the event may have happened minutes or days ago, and the `context` present in Configuration _now_ is likely not related.

To correctly handle both, I'd recommend setting context for Case 1 within the Event constructor used by `notify()` and leaving this function as it was, as it parses the serialized event file.

Reply:
Understood, and thanks for the explanation.  I've reverted `BSGParseContext()` and set `context` in the Event constructor, based on metadata (if present) or Configuration (if present).